core/src/main/java/org/apache/druid/timeline/partition/PartialShardSpec.java

```diff
@@ -29,7 +29,10 @@
 
 /**
  * Class to contain all information of a {@link ShardSpec} except for the partition ID.
- * This class is mainly used by the indexing tasks to allocate new segments using the Overlord.
+ * This class is used when the segment allocation is coordinated by the Overlord; when appending segments to an
+ * existing datasource (either streaming ingestion or batch append) or when using segment locking.
```

Comment:
Is this used for all segment locking cases? (is PartialShardSpec also involved in overwrites with segment locking, or is that just OverwriteShardSpec?)

Reply:
Yes, it's used all cases when segment lock is used. `PartialShardSpec` is used in the segment allocation protocol when the overlord coordinates the allocation. So, it is involved in 1) all cases with segment locking and 2) appending with time chunk locking. `OverwriteShardSpec` is a special shard spec used with segment locking for the segments overwriting others. Added some more details.